modules/features2d/src/fast.cpp

```diff
@@ -417,6 +418,65 @@ static bool openvx_FAST(InputArray _img, std::vector<KeyPoint>& keypoints,
 
 #endif
 
+namespace hal
+{
+inline int FAST(cv::Mat& src, std::vector<KeyPoint>& keypoints, int threshold, bool nonmax_suppression, int type)
+{
+    if (type != FastFeatureDetector::TYPE_9_16 || threshold > 20)
+        return CV_HAL_ERROR_NOT_IMPLEMENTED;
```

Comment:
@mshabunin, maybe we need to add `FAST_5_8` and `FAST_7_12` or FAST with parameter `type`, and check it inside HAL. What do you think?

Reply:
@ElenaGvozdeva , yes it will be better to define available types in `features2d/hal/interface.h` file (see [imgproc/hal/interface.h](https://github.com/opencv/opencv/blob/master/modules/imgproc/include/opencv2/imgproc/hal/interface.h) for example).